[FLINK-17976][docs][k8s/docker] Improvements about custom docker images (#12558)
apache/flink: docs/ops/deployment/index.md

<span class="label label-primary">Azure</span>
<span class="label label-primary">Google Cloud</span>
<span class="label label-primary">On-Premise</span>

## Deployment Best Practices

### How to provide dependencies in the classpath

Flink provides several approaches for providing dependencies (such as `*.jar` files or static data) to Flink or user-provided
applications. These approaches differ based on the deployment mode and target, but also have commonalities, which are described here.

To provide a dependency, there are the following options:

> - files in the **`lib/` folder** are added to the classpath used to start Flink. It is suitable for libraries such as Hadoop or file systems not available as plugins. Beware that classes added here can potentially interfere with Flink, for example if Flink you are adding a different version of a library already provided by Flink.

Reviewer: ```suggestion
- files in the **`lib/` folder** are added to the classpath used to start Flink. It is suitable for libraries such as Hadoop or file systems not available as plugins. Beware that classes added here can potentially interfere with Flink, for example if you are adding a different version of a library already provided by Flink.
```

- **`plugins/<name>/`** are loaded at runtime by Flink through separate classloaders to avoid conflicts with classes loaded and used by Flink. Only jar files which are prepared as [plugins]({% link ops/plugins.md %}) can be added here.

### Download Maven dependencies locally

If you need to extend the Flink with a Maven dependency (and its transitive dependencies),

> you can use a [Apache Maven](https://maven.apache.org) *pom.xml* file such as to download all required files into a local folder:

Reviewer: ```suggestion
you can use an [Apache Maven](https://maven.apache.org) *pom.xml* file to download all required files into a local folder:
```
